File: backend/src/main/java/com/team05/petmeeting/domain/adoption/repository/AdoptionApplicationRepository.java
package com.team05.petmeeting.domain.adoption.repository;

import com.team05.petmeeting.domain.adoption.entity.AdoptionApplication;
import com.team05.petmeeting.domain.adoption.entity.AdoptionStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.Optional;

@Repository
public interface AdoptionApplicationRepository extends JpaRepository<AdoptionApplication, Long> {
    List<AdoptionApplication> findByUser_Id(Long userId);
    Optional<AdoptionApplication> findByIdAndUser_Id(Long applicationId, Long userId);
    boolean existsByUser_IdAndAnimal_Id(Long userId, Long animalId);
    List<AdoptionApplication> findByAnimal_Shelter_CareRegNo(String careRegNo);
    List<AdoptionApplication> findByUser_IdAndStatus(Long userId, AdoptionStatus status);
}

File: backend/src/main/java/com/team05/petmeeting/domain/adoption/repository/AdoptionApplicationRepository.kt
package com.team05.petmeeting.domain.adoption.repository

import com.team05.petmeeting.domain.adoption.entity.AdoptionApplication
import com.team05.petmeeting.domain.adoption.entity.AdoptionStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository


@Repository
interface AdoptionApplicationRepository : JpaRepository<AdoptionApplication, Long> {
    fun findByUser_Id(userId: Long): List<AdoptionApplication>
    fun findByIdAndUser_Id(applicationId: Long, userId: Long): java.util.Optional<AdoptionApplication>
    fun existsByUser_IdAndAnimal_Id(userId: Long, animalId: Long): Boolean
    fun findByAnimal_Shelter_CareRegNo(careRegNo: String): List<AdoptionApplication>
    fun findByUser_IdAndStatus(userId: Long, status: AdoptionStatus): List<AdoptionApplication>

    fun existsByUser_IdAndAnimal_IdAndStatus(
        userId: Long,
        animalId: Long,
        status: AdoptionStatus
    ): Boolean
}
